epoch.go

```diff
@@ -877,41 +877,15 @@ func (e *Epoch) hasSomeNodeSignedTwice(nodeIDs []NodeID) bool {
 	return false
 }
 
-func (e *Epoch) handleBlockMessage(message *BlockMessage, from NodeID) error {
-	block := message.Block
-	if block == nil {
-		e.Logger.Debug("Got empty block in a BlockMessage")
-		return nil
-	}
-
+// handleBlockMessageFromLeader expects the block message
+// to come from the leader of the round, otherwise the block will not be processed.
+func (e *Epoch) handleBlockMessageFromLeader(message *BlockMessage, from NodeID) error {
```

Comment:
probably this method should be called `handleBlockMessage` while the other one should be called `processBlock` or something like that, and have its arguments changed to not accept a block message but just the block.